src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -228,9 +229,17 @@ private static void WriteHttpMethods(
                 .Properties
                 .FirstOrDefault(static prop => prop.IsOfKind(CodePropertyKind.Headers));
 
-            if (authenticationMethod != null && Enum.IsDefined(typeof(Authentication), authenticationMethod.Type.Name))
+            if (authenticationMethod != null && Enum.TryParse(typeof(SecuritySchemeType), authenticationMethod.Type.Name, true, out var _))
             {
-                writer.WriteLine($"Authorization: {{{{{authenticationMethod.DefaultValue}}}}}");
+                var schemeTypeMapping = new Dictionary<string, string>
```

Comment:
this dictionary is effectively a constant, so please move it outside of the method body to avoid unnecessary allocs. Or use a switch instead.